python/llm/example/GPU/HF-Transformers-AutoModels/Model/phi-3-vision/generate.py

```diff
@@ -0,0 +1,100 @@
+
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+    image_path = args.image_url_or_path
+
+    # Load model in INT8 for better output,
+    # which convert the relevant layers in the model into INT8 format
+    # When running LLMs on Intel iGPUs for Windows users, we recommend setting `cpu_embedding=True` in the from_pretrained function.
+    # This will allow the memory-intensive embedding layer to utilize the CPU instead of iGPU.
+    # `_attn_implementation="eager"` is required for phi-3-vision
+    # `modules_to_not_convert=["vision_embed_tokens"]` and `model = model.half()` are for acceleration and are optional
+    model = AutoModelForCausalLM.from_pretrained(model_path,
+                                                 trust_remote_code=True,
+                                                 load_in_low_bit="sym_int8",
+                                                 _attn_implementation="eager",
+                                                 modules_to_not_convert=["vision_embed_tokens"]).eval()
+    model = model.half().to('xpu')
+    
+    # Load tokenizer
+    processor = AutoProcessor.from_pretrained(model_path, trust_remote_code=True)
+    
+    messages = [
+        {"role": "user", "content": "<|image_1|>\n{prompt}".format(prompt=args.prompt)},
+    ]
+    prompt = processor.tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
+
+    if os.path.exists(image_path):
+       image = Image.open(image_path)
+    else:
+       image = Image.open(requests.get(image_path, stream=True).raw)
+    
+    # Generate predicted tokens
+    with torch.inference_mode():
+        # ipex_llm model needs a warmup, then inference time can be accurate
+        inputs = processor(prompt, [image], return_tensors="pt")
+        inputs = inputs.to('xpu')
+        output = model.generate(**inputs,
+                                eos_token_id=processor.tokenizer.eos_token_id,
+                                num_beams=1,
+                                do_sample=False,
+                                max_new_tokens=args.n_predict,
+                                temperature=0.0)
+        # start inference
+        st = time.time()
+        
+        inputs = processor(prompt, [image], return_tensors="pt")
+        inputs = inputs.to('xpu')
+        output = model.generate(**inputs,
+                                eos_token_id=processor.tokenizer.eos_token_id,
+                                num_beams=1,
+                                do_sample=False,
+                                max_new_tokens=args.n_predict,
+                                temperature=0.0)
+        end = time.time()
+        print(f'Inference time: {end-st} s')
+        output_str = processor.decode(output[0],
+                                      skip_special_tokens=True,
+                                      clean_up_tokenization_spaces=False)
+        print('-'*20, 'Prompt', '-'*20)
+        print(messages)
```

Comment:
Let's show the input image/url as well

Refer to: https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/example/GPU/HF-Transformers-AutoModels/Model/qwen-vl#sample-chat

Reply:
Let's also print the image link/path when showing the prompt?